app/templating/schema_context.py

```diff
@@ -1,25 +1,43 @@
+from app.questionnaire_state.state_repeating_answer_question import iterate_over_instance_ids
 from app.utilities.date_utils import to_date
 
 
-def build_schema_context(metadata, aliases, answers):
+def build_schema_context(metadata, aliases, answers, group_instance=None):
     """
     Build questionnaire schema context containing exercise and answers
     :param metadata: user metadata
     :param aliases: alias mapping of friendly names to answer ids
     :param answers: all the answers for the given questionnaire
+    :param group_instance: The group instance Id passed into the url route
     :return: questionnaire schema context
     """
     return {
         "exercise": _build_exercise(metadata),
         "answers": _map_alias_to_answers(aliases, answers),
+        "group_instance": group_instance,
     }
 
 
 def _map_alias_to_answers(aliases, answers):
     values = {}
     for alias, item_id in aliases.items():
-        value = answers.get(item_id)
-        values[alias] = value if value else ""
+        matching_answers = []
+        for answer_id, answer_index in iterate_over_instance_ids(answers):
+            if answer_id == item_id:
```

Comment:
Maybe we should pass the answer_store instead of the flattened answers. Then we could just filter for answers with the same item_id otherwise we are implementing filtering logic on top of the flattened answers